// @danitech/scraper/index.js
const scraper = require('./lib/scraper');
module.exports = scraper;

/*
module.exports = {
  downloader: {
    youtube_video: downloader_youtube_video,
    facebook: downloader_facebook,
    twitter: downloader_twitter,
    instagram: downloader_instagram,
    tiktok: downloader_tiktok,
    tiktok_v2: downloader_tiktok_v2,
    google_drive: downloader_google_drive
  },
  searcher: {
    lyrics: searcher_lyrics,
    soundcloud: searcher_soundcloud,
    emoji: searcher_emoji,
    stackoverflow: searcher_stackoverflow
  },
  stalker: {
    tiktok: stalker_tiktok,
    github_user: stalker_github_user,
    github_repo: stalker_github_repo,
    github_show_all_repo: stalker_github_show_all_repo,
    npmjs_package: stalker_npmjs_package,
    ip: stalker_ip
  },
  artificial_intelligence: {
    chatgpt_3: artificial_intelligence_openai_chatgpt_3,
    chatgpt_35: artificial_intelligence_openai_chatgpt_35,
    chatty_ai: artificial_intelligence_chatty_ai,
    bard: artificial_intelligence_google_bard,
    midjourney: artificial_intelligence_midjourney,
    anime_filter: artificial_intelligence_anime_filter
  },
  random_image: {
    anime_sfw: {
      akira: random_image_anime_sfw_akira,
      akiyama: random_image_anime_sfw_akiyama,
      ana: random_image_anime_sfw_ana,
      asuna: random_image_anime_sfw_asuna,
      ayuzawa: random_image_anime_sfw_ayuzawa,
      boruto: random_image_anime_sfw_boruto,
      chitanda: random_image_anime_sfw_chitanda,
      chitoge: random_image_anime_sfw_chitoge,
      cosplay: random_image_anime_sfw_cosplay,
      deidara: random_image_anime_sfw_deidara,
      doraemon: random_image_anime_sfw_doraemon,
      elaina: random_image_anime_sfw_elaina,
      emilia: random_image_anime_sfw_emilia,
      erza: random_image_anime_sfw_erza,
      fanart: random_image_anime_sfw_fanart,
      genshin: random_image_anime_sfw_genshin,
      gremory: random_image_anime_sfw_gremory,
      hestia: random_image_anime_sfw_hestia,
      hinata: random_image_anime_sfw_hinata,
      husbu: random_image_anime_sfw_husbu,
      waifu: random_image_anime_sfw_waifu,
      icon: random_image_anime_sfw_icon,
      inori: random_image_anime_sfw_inori,
      isuzu: random_image_anime_sfw_isuzu,
      itachi: random_image_anime_sfw_itachi,
      itori: random_image_anime_sfw_itori,
      kaga: random_image_anime_sfw_kaga,
      kagura: random_image_anime_sfw_kagura,
      kaguya: random_image_anime_sfw_kaguya,
      kakasih: random_image_anime_sfw_kakasih,
      kaneki: random_image_anime_sfw_kaneki,
      kaori: random_image_anime_sfw_kaori,
      keneki: random_image_anime_sfw_keneki,
      kosaki: random_image_anime_sfw_kosaki,
      kotori: random_image_anime_sfw_kotori,
      kuriyama: random_image_anime_sfw_kuriyama,
      kuroha: random_image_anime_sfw_kuroha,
      kurumi: random_image_anime_sfw_kurumi,
      loli: random_image_anime_sfw_loli,
      madara: random_image_anime_sfw_madara,
      menus: random_image_anime_sfw_menus,
      mikasa: random_image_anime_sfw_mikasa,
      miku: random_image_anime_sfw_miku,
      minato: random_image_anime_sfw_minato,
      naruto: random_image_anime_sfw_naruto,
      natsukawa: random_image_anime_sfw_natsukawa,
      neko: random_image_anime_sfw_neko,
      nekonime: random_image_anime_sfw_nekonime,
      nezuko: random_image_anime_sfw_nezuko,
      nishimiya: random_image_anime_sfw_nishimiya,
      onepiece: random_image_anime_sfw_onepiece,
      pokemon: random_image_anime_sfw_pokemon,
      rem: random_image_anime_sfw_rem,
      rize: random_image_anime_sfw_rize,
      sagiri: random_image_anime_sfw_sagiri,
      sakura: random_image_anime_sfw_sakura,
      sasuke: random_image_anime_sfw_sasuke,
      shina: random_image_anime_sfw_shina,
      shinka: random_image_anime_sfw_shinka,
      shizuka: random_image_anime_sfw_shizuka,
      shota: random_image_anime_sfw_shota,
      simp: random_image_anime_sfw_simp,
      tomori: random_image_anime_sfw_tomori,
      toukachan: random_image_anime_sfw_toukachan,
      tsunade: random_image_anime_sfw_tsunade,
      yatogami: random_image_anime_sfw_yatogami,
      yuki: random_image_anime_sfw_yuki
    },
    anime_nsfw: {
      ahegao: random_image_anime_nsfw_ahegao,
      ass: random_image_anime_nsfw_ass,
      bdsm: random_image_anime_nsfw_bdsm,
      blowjob: random_image_anime_nsfw_blowjob,
      cuckold: random_image_anime_nsfw_cuckold,
      cum: random_image_anime_nsfw_cum,
      eba: random_image_anime_nsfw_eba,
      ero: random_image_anime_nsfw_ero,
      femdom: random_image_anime_nsfw_femdom,
      foot: random_image_anime_nsfw_foot,
      gangbang: random_image_anime_nsfw_gangbang,
      gifs: random_image_anime_nsfw_gifs,
      glasses: random_image_anime_nsfw_glasses,
      hentai: random_image_anime_nsfw_hentai,
      jahy: random_image_anime_nsfw_jahy,
      manga: random_image_anime_nsfw_manga,
      masturbation: random_image_anime_nsfw_masturbation,
      megumin: random_image_anime_nsfw_megumin,
      neko: random_image_anime_nsfw_neko,
      nekonime: random_image_anime_nsfw_nekonime,
      loli: random_image_anime_nsfw_loli,
      manga: random_image_anime_nsfw_manga,
      orgy: random_image_anime_nsfw_orgy,
      panties: random_image_anime_nsfw_panties,
      pussy: random_image_anime_nsfw_pussy,
      tentacles: random_image_anime_nsfw_tentacles,
      thighs: random_image_anime_nsfw_thighs,
      yuri: random_image_anime_nsfw_yuri,
      zettai: random_image_anime_nsfw_zettai
    }
  },
  converter: {
    text_to_image: converter_text_to_image,
    text_to_gif: converter_text_to_gif
  },
  maker: {
    enhance: maker_enhance,
    beautiful: maker_beautiful,
    blur: maker_blur,
    facepalm: maker_facepalm,
    invert: maker_invert,
    rainbow: maker_rainbow,
    trigger: maker_trigger,
    wanted: maker_wanted,
    wasted: maker_wasted,
    darkness: maker_darkness,
    pixelate: maker_pixelate
  },
  tools: {
    get_temp_mail: tools_get_temp_mail,
    get_temp_mail_inbox: tools_get_temp_mail_inbox,
    screenshot_website,
    style_text,
    base64_encode,
    base64_decode,
    base32_encode,
    base32_decode
  }
};
*/
